### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by brighton0102**
In this milestone We : 

- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components:
- Use Jest snapshots to test the components.
- Use React Testing Library to simulate user interaction.

### CODE REVIEW

File: src/components/Button.test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import renderer from 'react-test-renderer';
```
Comment (V-Blaze) [inline src/components/Button.test.js]: - You have done great so far :+1: :+1: Your app meets all the requirements and all tests passed as expected. but there are a few things you need to correct before moving to the next milestone. :smile: :wave: 

- [ ] As required by Microverse, you are not meant to use any linters disable rule in your code base. The linters are there to make sure your code follows the best industry standard. I'll suggest you remove this `disable` rule and try to fix the issue. :smile: :small_airplane:

File: src/components/Calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (V-Blaze) [inline src/components/Calculator.test.js]: - [ ] As I have pointed out above, you are not meant to use any linters disable rule in your code base. The linters are there to make sure your code follows the best industry standard. I'll suggest you remove this `disable` rule and try to fix the issue. :smile: :small_airplane:

File: src/components/Home.test.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import renderer from 'react-test-renderer';
```
Comment (V-Blaze) [inline src/components/Home.test.js]: - [ ] I'll suggest you remove this too and try to fix the issue :+1: :100:

File: src/components/QuoteDisplay.test.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import renderer from 'react-test-renderer';
```
Comment (V-Blaze) [inline src/components/QuoteDisplay.test.js]: - [ ] I'll suggest you remove this too and try to fix the issue :+1: :100:

File: src/components/QuoteDisplay.test.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+import QuoteDisplay from './QuoteDisplay';
+
+test('QuoteDisplay snapshot test', () => {
+  const component = renderer.create(<QuoteDisplay />);
+  const tree = component.toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (V-Blaze) [inline src/components/QuoteDisplay.test.js:9]: - [ ] You have a failed test coming from `App.test.js`. this is a boilerplate test file from React, I suggest you remove it. :wave: 

- - Screenshot reference:
Screenshot from 2023-09-29 09-38-54

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @brighton0102, @Goldinium  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Clean and well organised codebase✔
-  All tests are passing👋 
-  Used jest for snapshot testing✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — APPROVED** [review]
# Hi @brighton0102 & @Goldinium 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
image
- All tests are passing ✔
- Linter disable rules are removed ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**